### DISCUSSION TITLE
Can this upgrade work with Gaggia Espresso Evolution?

### DISCUSSION THREAD

**Discussion by thanaman** [Zer0-bit/gaggiuino#222] [General]
Can this upgrade work with Gaggia Espresso Evolution?, which parts should I buy GC or GCP for extended functionality

Thank you

**Comment by NicholasGrammaticus** [comment]
There is no fundamental difference between the Evolution and the Classic, so this will work if you can figure out what you might need to do differently. 
The GCP has extra terminals on the brew switch. The GC uses compression fittings while the GCP uses barb fittings. Take a look inside your machine and verify this but I believe you have compression fittings like the GC, and you will probably need the optocoupler like the GC.
